src/main/java/com/sap/oss/phosphor/fosstars/model/value/ArtifactVersion.java

```diff
@@ -0,0 +1,132 @@
+package com.sap.oss.phosphor.fosstars.model.value;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.core.JsonGenerator;
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.SerializerProvider;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.fasterxml.jackson.databind.annotation.JsonSerialize;
+import com.fasterxml.jackson.databind.deser.std.StdDeserializer;
+import com.fasterxml.jackson.databind.ser.std.StdSerializer;
+import java.io.IOException;
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Contains version tag and release date for an artifact version.
+ */
+public class ArtifactVersion {
```

Comment:
Are you planning to add support for not-semantic versions?

Reply:
The semantic version are used for the version ranges in the CVEs.
For the other version related scores the semantic version is not used.
Here the direct version string is used to find the correct version.
Hence, the version is used in the form it is provided and the semantic version in addition for version ranges.

Reply:
There are projects that don't use semantic versions, and besides CVE there are other vulnerability identifiers. It looks like that the new rating procedure is supposed to work only with semantic versions and CVE identifiers. Is it by design?

Reply:
Yes, the `ArtifactVersionVulnerabilityScore` only works with CVEs and semantic versions.
To improve this the issue #480 was created.
But the other scores like `ArtifactReleaseHistoryScore` does not require a semantic version.